### PR TITLE
6.0.0 dev

### LINKED ISSUE #149: Upgrade to Symfony 3.x
Since tests are failing because Twig deprecated a bunch of features used in Symfony 2.8, I think it's time we upgrade Symfony

### PR DESCRIPTION
**Pull request by ohvitorino**
This upgrades the Framework's Symfony version to 3.3 and it's associated dependencies.

Closes #149

### CODE REVIEW

**Comment by jonasdekeukelaere** [comment]
@ohvitorino can you merged master in this branch?

**Comment by ohvitorino** [comment]
@jonasdekeukelaere I have merged master into this branch and on the way I also fixed some issues in the examples and the `FieldsetType`.

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+dist: precise
```
Comment (tijsverkoyen) [inline .travis.yml]: Why? Are the tests failing?

File: .travis.yml [hunk 2]
```diff
@@ -20,7 +22,7 @@ before_script:
 #  - php app/console doctrine:fixtures:load --env=test --no-interaction
 
   # install required gulp-packages
-  - npm install gulp@"^3.9.0" gulp-rename@"^1.2.2" gulp-coffee@"^2.3.1" gulp-util@"^3.0.6" -g
+  - npm install gulp@"^3.9.0" gulp-rename@"^1.2.2" gulp-coffee@"^2.3.1" gulp-util@"^3.0.6"
```
Comment (tijsverkoyen) [inline .travis.yml]: Wouldn't it be beter to use yarn? See https://blog.travis-ci.com/2016-11-21-travis-ci-now-supports-yarn
With yarn, we can use our yarn.lock-file and make sure we have the correct dependencies installed.

File: .travis.yml [hunk 3]
```diff
@@ -8,15 +8,7 @@ language: php
...
-    - php: 5.5
-    - php: 5.6
-    - php: 7
-    - php: nightly
-    - php: hhvm
-  allow_failures:
-    - php: 5.5
-    - php: nightly
-    - php: hhvm
+    - php: 7.1
```
Comment (tijsverkoyen) [inline .travis.yml]: Shouldn't we support `nightly`? But allow failures, with that we can detect issues before a new version is released.

File: app/config/config.yml [hunk 4]
```diff
@@ -27,29 +27,31 @@ framework:
...
   templating:
     engines: ['twig']
   default_locale:  "%locale%"
   trusted_hosts:   ~
-  trusted_proxies: ~
   fragments:       ~
   assets:
     version: 1
     version_format: '%%s?v=%%s'
+  php_errors:
```
Comment (tijsverkoyen) [inline app/config/config.yml]: Shouldn't this be in the config.dev.yml file?

File: app/config/config_dev.yml [hunk 5]
```diff
@@ -23,6 +23,16 @@ monolog:
 #        chromephp:
 #            type:  chromephp
 #            level: info
+        console:
+          type: console
+          process_psr_3_messages: false
+          channels: ['!event', '!doctrine', '!console']
+        # To follow logs in real time, execute the following command:
```
Comment (tijsverkoyen) [inline app/config/config_dev.yml]: I would add the comments above `console:`

File: .travis.yml (hunk 1 above)
Comment (ohvitorino) [inline .travis.yml]: Yup, travis setup fails if we don't force the previous ubuntu version. See #184.

File: .travis.yml (hunk 2 above)
Comment (ohvitorino) [inline .travis.yml]: Correct. I'll change this to yarn.

File: .travis.yml (hunk 3 above)
Comment (ohvitorino) [inline .travis.yml]: It's a good idea. I'll also add nightly to the list.

File: .travis.yml (hunk 1 above)
Comment (tijsverkoyen) [inline .travis.yml]: k

File: .travis.yml (hunk 2 above)
Comment (ohvitorino) [inline .travis.yml]: 8383c6f adds yarn support and installation.

File: .travis.yml (hunk 3 above)
Comment (ohvitorino) [inline .travis.yml]: 5f374cf adds PHP's nightly build to travis configuration.

File: app/config/config.yml (hunk 4 above)
Comment (ohvitorino) [inline app/config/config.yml]: @jonasdekeukelaere can you give some input on this?

Comment (jonasdekeukelaere) [inline app/config/config.yml]: copied from the symfony-standard https://github.com/symfony/symfony-standard/blob/3.3/app/config/config.yml#L33 - looks like it's to log the php errors

File: app/config/config_dev.yml (hunk 5 above)
Comment (ohvitorino) [inline app/config/config_dev.yml]: I moved the comments to just underneath the `console` key. I think this way it is more clear that it concerns the console and it's more visible.

File: app/config/config.yml (hunk 4 above)
Comment (ohvitorino) [inline app/config/config.yml]: Ok, so I guess PHP errors will also be logged to the application log file.

Comment (jonasdekeukelaere) [inline app/config/config.yml]: I haven't looked into it, but I guess yes

Comment (ohvitorino) [inline app/config/config.yml]: I would actually leave this parameter in `config.yml`, I just don't know how errbit handles this.

Comment (tijsverkoyen) [inline app/config/config.yml]: ok.

File: app/config/config_dev.yml (hunk 5 above)
Comment (tijsverkoyen) [inline app/config/config_dev.yml]: In all other places we add comments above the item, so I don't get why this should not follow that convention.
But no deal breaker for me.

Comment (ohvitorino) [inline app/config/config_dev.yml]: Ok, it seems fair. I didn't have a look at the other config files. It just seamed more clear just under `console`.
Fixed 49abb52.

**Pull request merged**